### PR TITLE
Write checkpoints to Checkpoints/ directory

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Preparation for #4996.

### Upgrade instructions

Checkpoints are now written in the format `Checkpoints/Checkpoint_0123`.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: tests/InputFiles/ExampleExecutables/RandomAmr1D.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ Executable: RandomAmr1D
 Testing:
   Check: parse;execute_check_output
 ExpectedOutput:
- - SpectreCheckpoint000000
+  - Checkpoints/Checkpoint_000000
```
Comment (knelli2) [inline tests/InputFiles/ExampleExecutables/RandomAmr1D.yaml]: Too many zeros?

**Review by knelli2 — CHANGES_REQUESTED** [review]
Small thing

File: tests/InputFiles/ExampleExecutables/RandomAmr1D.yaml (hunk 1 above)
Comment (nilsvu) [reply tests/InputFiles/ExampleExecutables/RandomAmr1D.yaml]: Fixed. There's a bug in the CMake testing code that I'll try to fix in a separate PR. Edit: #5080

**Review by knelli2 — APPROVED** [review]

**Comment by nilsvu** [comment]
@wthrowe could you take this one as well?

**Review by wthrowe — DISMISSED** [review]

**Comment by wthrowe** [comment]
Approved, but needs a rebase.

**Comment by nilsvu** [comment]
done

**Review by wthrowe — APPROVED** [review]

**Pull request merged**